components/discord_bot/discord_bot.app.mjs

```diff
@@ -0,0 +1,530 @@
+      type: "integer",
+      label: "Position",
+      description: "The position of the channel in the left-hand listing",
+      optional: true,
+    },
+    channelTopic: {
+      type: "string",
+      label: "Topic",
+      description: "0-1024 character channel topic",
+      optional: true,
+    },
+    channelNsfw: {
+      type: "boolean",
+      label: "NSFW",
+      description: "[Not Safe For Wumpus](https://support.discord.com/hc/en-us/articles/115000084051-NSFW-Channels-and-Content)",
+      optional: true,
+    },
+    channelBitrate: {
+      type: "integer",
+      label: "Bitrate",
+      description: "The bitrate (in bits) of the voice channel; 8000 to 96000 (128000 for VIP servers).",
+      optional: true,
+    },
+    channelRateLimitPerUser: {
+      type: "integer",
+      label: "Rate limit per user",
+      description: "Amount of seconds a user has to wait before sending another message (0-21600); bots, as well as users with the permission manage_messages or manage_channel, are unaffected.",
+      optional: true,
+    },
+    channelUserLimit: {
+      type: "integer",
+      label: "User limit",
+      description: "The user limit of the voice channel; 0 refers to no limit, 1 to 99 refers to a user limit.",
+      optional: true,
+    },
+    channelParentId: {
+      type: "string",
+      label: "Category",
+      description: "Id of the new parent category for a channel",
+      optional: true,
+      async options({ guildId }) {
+        const channels = await this.getGuildChannels({
+          guildId,
+        });
+        return utils.getCategoryChannelOptions(channels);
+      },
+    },
+    channelRolePermissions: {
+      type: "string[]",
+      label: "Roles to add",
+      description: "Choose the roles you want to add to your channel",
+      optional: true,
+      async options({ guildId }) {
+        const roles = await this.getGuildRoles({
+          guildId,
+        });
+        return utils.getRolePermissionOptions(roles);
+      },
+    },
+    channelMemberPermissions: {
+      type: "string[]",
+      label: "Members to add",
+      description: "Choose the members you want to add to your channel",
+      optional: true,
+      async options({
+        guildId, prevContext,
+      }) {
+        const { after } = prevContext || {};
+        const responses = await Promise.all([
+          this.getGuild({
+            guildId,
+          }),
+          this.getGuildRoles({
+            guildId,
+          }),
+          this.getGuildMembers({
+            guildId,
+            after,
+            limit: 100,
+          }),
+        ]);
+
+        const [
+          ,, members,
+        ] = responses;
+
+        const lastMemberId = utils.getLastId(members);
+        const options = utils.getMemberPermissionOptions(responses);
+
+        return {
+          options,
+          context: {
+            after: lastMemberId,
+          },
+        };
+      },
+    },
+    max: {
```

Comment:
@jcortes shouldn't `max` be optional?

Reply:
Yeah I agree, let me do the change then!